### ISSUE TITLE
429 Too many requests

### ISSUE THREAD

**Issue by atpanos** [TobseF/impf-bot#40]
Is it only me?
After some small amount of time I get only 429 on the termincheck.
Any workaround?
I did not change any time parameter.

**Comment by Pfuenzle** [comment]
Nope, Me and some other people (https://github.com/TobseF/impf-bot/issues/11) also have this issue, it's probably because of the bot entering the code too fast.
@prok42 had the idea to use the URL that came with the confirmation email, as this way the entry of the code can be bypassed, which will probably fix this

**Comment by atpanos** [comment]
I don't even have a confirmation code. Still trying to get one for the age of 62.

**Comment by atpanos** [comment]
![Screenshot_20210515-163603.png](https://user-images.githubusercontent.com/26478584/118365114-ac49ad00-b59b-11eb-92fc-adff51a2eaed.png)
I set the timer up to 300 and still the same issue...

**Comment by KingLoui1986** [comment]
I also experiance the same, I also played around with different time settings.
But the webpage seems to identify my connection attempt immediatly as a bot attempt. I have no other explanation for it than that.
It also recognizes my very first connection attempt as a Bot and  immediatly replys with 429.
Does no one else have a similar same observation?
Did anyone found a solution for the problem already?
![image](https://user-images.githubusercontent.com/84266125/118557895-2594fb80-b766-11eb-94bc-ec4127f618e1.png)

**Comment by anp369** [comment]
I also have the same problem. Reloading the page by hand and entering the code by hand fixes the problem four roughly 5-10 minutes before the 429 erros pop up again. Strangely this depends on the impfzentrum. Some in my list stop working while others continue to work for some more requests.

**Comment by KingLoui1986** [comment]
I think this suggestion posted by @TobseF in Issue #43 could potentially be a solution for this problem as well.

> Ich weiß nicht wie es sich auf die Suche auswirkt, aber wenn bei jedem Check ein "neuer Browser" genommen werden soll, dann lässt sich dass sehr einfach nachrüsten. Vielleicht wird das Cookie mitverwendet um häufige Abfragen zu erkennen. Dann würde es helfen in der `locations`-Schleife den Driver neu aufzubauen:

```kotlin
driver.close()
driver = createDriver()
```

_Originally posted by @TobseF in https://github.com/TobseF/impf-bot/issues/43#issuecomment-843459995_

**Comment by atpanos** [comment]
i don't think it will have any affect.
i just ran it on windows (gradle build, java -jar ...) and it works.
only on docker on my raspberry i get only 429, even on the first call.

**Comment by Pfuenzle** [comment]
Did you try to change the useragent to a windows one?

**Comment by atpanos** [comment]
i did, but somehow it does not work, you mean through the config file, correct?

**Comment by Pfuenzle** [comment]
Yes, with the config, the server should then have no idea if the client is running in docker or on a desktop

**Comment by atpanos** [comment]
#61 
there was a change required to get the change of user-agent to work.
on docker in windows now it works.
now i will try on raspberry pi.

EDIT: still not working on raspberry pi :(

**Comment by atpanos** [comment]
@KingLoui1986 did you try the driver close solution?

**Comment by TimApplepie** [comment]
> @prok42 had the idea to use the URL that came with the confirmation email, as this way the entry of the code can be bypassed, which will probably fix this

I did exactly this in my [fork](https://github.com/TimApplepie/impf-bot/commit/c6b785088e09af190fe3c9219ab3e0b492495de4). Fixed the issue for me.
Feel free to use it.

**Comment by GameScripting** [comment]
@TimApplepie I tried to merge & use your changes, however I noticed that your change requires me to already have received a placement code, right?

**Comment by TimApplepie** [comment]
> @TimApplepie I tried to merge & use your changes, however I noticed that your change requires me to already have received a placement code, right?

Correct

**Comment by prok42** [comment]
@TimApplepie if you have the time, a pull request to get your feature merged here would be great (maybe add a switch how Vermuttlungscodes should be used - entered "interactively" or used "directly" (the latter may save you from 429, the first is maybe more robust if URL structures should change)
